osbenchmark/client.py

```diff
@@ -282,23 +281,35 @@ def parse_aws_log_in_params(self):
 
     def create(self):
         # pylint: disable=import-outside-toplevel
-        import opensearchpy
+        import osbenchmark.sync_connection
         from botocore.credentials import Credentials
 
+        if "connection_class" in self.client_options:
+            connection = self.client_options.pop("connection_class", None)
+            if connection.lower() in {"requestshttpconnection", "requests"}:
+                connection_class = osbenchmark.sync_connection.RequestsHttpConnection
+            else:
+                connection_class = osbenchmark.sync_connection.Urllib3HttpConnection
+        else:
+            connection_class = osbenchmark.sync_connection.Urllib3HttpConnection
+
```

Comment:
This seems unnecessarily complicated, and should raise an error if the user specified an invalid class.

```python
connection_class = self.client_options.get("connection_class", "urllib3")
if connection_class == "requests":
    connection_class = osbenchmark.sync_connection.RequestsHttpConnection
elif connection_class == "..."
   ...
else:
    throw ...
```


```

Reply:
> This seems unnecessarily complicated, and should raise an error if the user specified an invalid class.
> 
> ```python
> connection_class = self.client_options.get("connection_class", "urllib3")
> if connection_class == "requests":
>     connection_class = osbenchmark.sync_connection.RequestsHttpConnection
> elif connection_class == "..."
>    ...
> else:
>     throw ...
> ```

- Error handling is done [here](https://github.com/opensearch-project/opensearch-benchmark/pull/445/files#diff-c89a0ed9511e7ad75cf883fcc7c10bc472902bcb8d3b9f1ba86f2eb50a560636R1460) 
- I will simplify the code

Reply:
Agree with @dblock   Since Python does not have a `switch` or `case` statement, at least pre-3.10, you can use a structured if-elif block or perhaps a map.  Also, a `default` arm with error handling is good practice, even if there is error handling elsewhere, since a reader of this section might not be aware.